query/logicalplan/logicalplan.go

```diff
@@ -108,30 +117,21 @@ type TableReader interface {
 		tx uint64,
 		pool memory.Allocator,
 		schema *arrow.Schema,
-		physicalProjection []Expr,
-		projection []Expr,
-		filter Expr,
-		distinctColumns []Expr,
+		options *IterOptions,
```

Comment:
Repeating the same comment as on the other PR: I think we can change this pointer argument to a value to avoid allocations